scripts/lib/pb_core_ingester.rb

```diff
@@ -81,14 +83,17 @@ def ingest_xml_no_commit(xml)
       raise ValidationError.new(e)
     end
 
-    begin
-      @solr.add(pbcore.to_solr)
-    rescue => e
-      raise SolrError.new(e)
+    if @re.match(xml)
```

Comment:
@afred: 
```
r.empty?
NoMethodError: undefined method `empty?' for //:Regexp
```
I wouldn't worry about the speed: The very first zero-width string before the first character will match, and that's that. It did feel a little sneaky to do this, but it seems better than passing along the original re-string.